Task1 c#

Syed-Ali-Abbas-568 opened the pull request.

Syed-Ali-Abbas-568 committed:
Task1 c#

Faisalimrann commented on Task1-C-Sharp.cs:
You did a great job. There are a few things I want you to look into.
1- How can we divide the code into multiple files? What are the best practices for it?
2- What if I write something after the line ```store.UpdateProductAvailability(chosenProduct.Name, daysUntilAvailable);``` Will it work?
3- It's a good practice to create a separate function for all validations. It cleans the code and makes it more readable.
4- Can you modify the code in such a way that the product has a release date? When the release date arrives the system notifies the interested customers.
5- If a user inputs the wrong value, should the program exit or ask for the correct input? What do you think is better as a software engineer?
6- Add a feature for customers who want to unsubscribe from a product.

Faisalimrann replied:
It is a good practice to include proper commit messages and a PR description when creating a PR. It would be better to create a branch using the JIRA ticket name and include the JIRA ticket in the PR description.

Syed-Ali-Abbas-568 committed:
Incorportated All PR comments and made neccessary changes

Syed-Ali-Abbas-568 commented:
JIRA ISSUE ID: https://arbisoft-team-ek6wvqc8.atlassian.net/browse/SAAAT-4
Events and Delegates Task of C# Practice.

Split Code into multiple files according to best practices


Q. What if I write something after the line store.UpdateProductAvailability(chosenProduct.Name, daysUntilAvailable);? Will it work?
Ans: Yes it will work but only after the method call has finished, ensuring that all updates and notifications have been processed before moving forward.

If the user enters a wrong input, instead of crashing the program should prompt the user to input the correct inputs.

Added the ability to unsubscribe from a notification.
